### DISCUSSION TITLE
Parsing Events from a Kafka topic with schema

### DISCUSSION THREAD

**Discussion by Srinu3366** [aws-powertools/powertools-lambda-python#3372] [General]
I am using Kafka event trigger with Confluent Kafka. I am getting an error when I am trying to use KafkaSelfManagedEventModel to parse the events. 

Kafka adds schema id as first five bytes to value for schema identification. KafkaSelfManagedEventModel tries to decode the bytes during parsing. It is not able to parse the first 5 bytes and giving me an error. 

I am wondering whether we need to create a separate parser for schema aware kafka events or any other option already available in powertools.

Error: 39 validation errors for KafkaSelfManagedEventModel\nrecords -> test-topic-3 -> 0 -> value\n  'utf-8' codec can't decode byte 0x86 in position 3: invalid start byte (type=value_error.unicodedecode)

**Comment by leandrodamascena** [comment]
Hey @Srinu3366 I know this is almost 2 years since you posted it, but Lambda added support for Schema Registry integration and Powertools now supports it:

https://docs.powertools.aws.dev/lambda/python/latest/utilities/kafka/
